tests/Doctrine/Tests/Common/Persistence/Mapping/AnnotationDriverTest.php

```diff
@@ -6,18 +6,38 @@
 use Doctrine\Common\Persistence\Mapping\Driver\AnnotationDriver;
 use Doctrine\Common\Persistence\Mapping\ClassMetadata;
 use Doctrine\Entity;
+use Doctrine\SubDirTestClass;
+use Doctrine\SymlinkedTestClass;
 use Doctrine\TestClass;
 
 class AnnotationDriverTest extends \PHPUnit_Framework_TestCase
 {
     public function testGetAllClassNames()
     {
+        $this->makeSubDirSymlink();
+
         $reader = new AnnotationReader();
         $driver = new SimpleAnnotationDriver($reader, [__DIR__ . '/_files/annotation']);
 
         $classes = $driver->getAllClassNames();
 
-        $this->assertEquals([TestClass::class], $classes);
+        $expectedClasses = [SubDirTestClass::class, SymlinkedTestClass::class, TestClass::class];
+        sort($expectedClasses);
+        sort($classes);
+
+        $this->assertEquals($expectedClasses, $classes);
+    }
+
+    private function makeSubDirSymlink()
+    {
+        $symlinkTarget = __DIR__ . '/_files/SubDirForSymlink';
+        $symlinkPath = __DIR__ . '/_files/annotation/SubDirForSymlink';
+        if (! is_link($symlinkPath)) {
+            if (file_exists($symlinkPath)) {
+                throw new \Exception(sprintf('Remove file/dir "%s".', $symlinkPath));
+            }
+            symlink($symlinkTarget, $symlinkPath);
```

Comment:
Creating symlink on windows requires "Run as administrator" permissions, then tests run fine. Otherwise it gives "1) Doctrine\Tests\Common\Persistence\Mapping\Annota
symlink(): Cannot create symlink, error code(1314)".